intake/source/csv.py

```diff
@@ -49,23 +49,25 @@ def __init__(self, urlpath, csv_kwargs=None, metadata=None, storage_options=None
         """
         self.path_as_pattern = path_as_pattern
         self.urlpath = urlpath
-        self._storage_options = storage_options
+        self._storage_options = storage_options or {}
         self._csv_kwargs = csv_kwargs or {}
-        self._dataframe = None
+        self._dask_df = None
+        self._files = None
+        self._schema = None
 
         super(CSVSource, self).__init__(metadata=metadata)
 
     def _set_pattern_columns(self, path_column):
```

Comment:
@martindurant Does this need to happen for pandas too?

Reply:
Interesting question. Pandas doesn't have this functionality now, and doesn't load multiple files in one go anyway. Some might in principle be relying on this dask behaviour even when only ever using read(), but it seems unlikely. Let's see how tricky it is to add.

Reply:
Not too tricky, assuming I got it right!